lib/recipe/bloc/recipe_bloc.dart

```diff
@@ -0,0 +1,322 @@
+import 'package:bloc/bloc.dart';
+import 'package:equatable/equatable.dart';
+
+import 'package:gibsonify/recipe/models/models.dart';
+import 'package:gibsonify/recipe/recipe.dart';
+
+part 'recipe_event.dart';
+part 'recipe_state.dart';
+
+class RecipeBloc extends Bloc<RecipeEvent, RecipeState> {
+  RecipeBloc() : super(const RecipeState()) {
+    on<RecipeAdded>(_onRecipeAdded);
+    on<RecipeNameChanged>(_recipeNameChanged);
+    on<RecipeNumberChanged>(_recipeNumberChanged);
+    on<RecipeVolumeChanged>(_recipeVolumeChanged);
+    on<RecipeStatusChanged>(_onRecipeStatusChanged);
+    on<IngredientAdded>(_onIngredientAdded);
+    on<IngredientStatusChanged>(_onIngredientStatusChanged);
+    on<IngredientNameChanged>(_onIngredientNameChanged);
+    on<IngredientDescriptionChanged>(_onIngredientDescriptionChanged);
+    on<CookingStateChanged>(_onCookingStateChanged);
+    on<MeasurementMethodChanged>(_onMeasurementMethodChanged);
+    on<MeasurementChanged>(_onMeasurementChanged);
+    on<MeasurementUnitChanged>(_onMeasurementUnitChanged);
+    on<SizeChanged>(_onSizeChanged);
+    on<SizeNumberChanged>(_onSizeNumberChanged);
+  }
+
+  void _onRecipeAdded(RecipeAdded event, Emitter<RecipeState> emit) {
+    final recipe = Recipe();
+    List<Recipe> recipes = List.from(state.recipes);
+    recipes.add(recipe);
+
+    emit(state.copyWith(recipes: recipes));
+  }
+
+  void _recipeNameChanged(RecipeNameChanged event, Emitter<RecipeState> emit) {
+    List<Recipe> recipes = List.from(state.recipes);
+
+    int changedRecipeIndex = recipes.indexOf(event.recipe);
```

Comment:
I don't think I completely follow the implementation you're hinting at but will refactor at a future point when I eventually see you implement an example

Reply:
Right now we are passing the whole instance of a `Recipe` in the `RecipeNameChanged` even and then acessing it via `event.recipe` just to perform a lookup for the index of that particular recipe (`recipes.indexOf(event.recipe);`) in the list of all recipes in the state (`state.recipes`). We are also passing the new `recipeName` in the event to than `copyWith` that recipe we looked up with the updated name.

This means we are passing around redundant data in the event — we are supplying the whole recipe, but we just need the information of which recipe to change and what should the new name be.

I think a cleaner implementation would be to just pass the UUID of the recipe we want to change and the new recipe name in the event, and then perform a lookup in `state.recipes` for the recipe that matches the UUID e.g. `recipes.where(recipe.id == changedRecipeId)`; I made up the syntax, but hopefully this illustrates what I mean 😄.

As you say, I'll do that when I refactor collection, and then it can be implemented here as well 😊.